### PR TITLE
Add API routes of chart repo services in harbor API router

### PR DESCRIPTION
**Pull request by steven-zou**
**Add API routes of chart repo services in harbor API router**

- add chart repo API controller
- update GetChartVersion to extract repo name from the request
- update the corresponding UI cases
- update the chart controller initialization process

### CODE REVIEW

File: src/ui/router.go [hunk 1]
```diff
@@ -122,6 +122,21 @@ func initRouters() {
 
 	beego.Router("/registryproxy/*", &controllers.RegistryProxy{}, "*:Handle")
 
+	//APIs for chart repository
+	chartRepositoryAPIType := &api.ChartRepositoryAPI{}
+	beego.Router("/api/chartserver/health", chartRepositoryAPIType, "get:GetHealthStatus")
+	beego.Router("/api/:repo/charts", chartRepositoryAPIType, "get:ListCharts")
```
Comment (reasonerjt) [inline src/ui/router.go]: Suggest we add a prefix to all the chart related APIs, for example `/api/chart_registry/:repo/charts`

File: src/ui/api/chart_repository.go [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+		if !cra.requireNamespace(cra.namespace) {
+			return
+		}
+	}
+}
+
+//GetHealthStatus handles GET /api/chartserver/health
+func (cra *ChartRepositoryAPI) GetHealthStatus() {
+	//Check access
+	cra.requireAccess(cra.namespace, accessLevelSystem)
```
Comment (reasonerjt) [inline src/ui/api/chart_repository.go]: should not move on if it returns false.

File: src/ui/router.go [hunk 3]
```diff
@@ -122,6 +122,21 @@ func initRouters() {
...
+	beego.Router("/api/:repo/charts/:name", chartRepositoryAPIType, "get:ListChartVersions")
+	beego.Router("/api/:repo/charts/:name/:version", chartRepositoryAPIType, "get:GetChartVersion")
+	beego.Router("/api/:repo/charts/:name/:version", chartRepositoryAPIType, "delete:DeleteChartVersion")
+	beego.Router("/api/:repo/charts", chartRepositoryAPIType, "post:UploadChartVersion")
+	beego.Router("/api/:repo/prov", chartRepositoryAPIType, "post:UploadChartProvFile")
+
+	//Repository services
+	beego.Router("/:repo/index.yaml", chartRepositoryAPIType, "get:GetIndexByRepo")
```
Comment (reasonerjt) [inline src/ui/router.go]: Same concern here, shall we add a prefix to isolate all requests for the charts?
I understand there maybe limitation as the request is from `helm` command. 
Maybe we can ask user to set `context` to `helm` command?
Let's open a follow-up issue for this concern.

File: src/ui/router.go (hunk 1 above)
Comment (steven-zou) [reply src/ui/router.go]: Maybe we can enable context

Comment (steven-zou) [reply src/ui/router.go]: Will find a proper way and submit code in separate PR

File: src/ui/api/chart_repository.go (hunk 2 above)
Comment (steven-zou) [reply src/ui/api/chart_repository.go]: Will be fixed

File: src/ui/router.go (hunk 3 above)
Comment (steven-zou) [reply src/ui/router.go]: Will find a proper way and submit code in separate PR

**Pull request merged**